shambomon/lib/shambomon/game.ex

```diff
@@ -172,42 +184,78 @@ defmodule Shambomon.Game do
     Map.put(game, :messages, msgs)
   end
 
+  defp update_messages(game, player, special_roll, _special?) do
+    msgs = Map.get(game, :messages)
+    # insert special attack related messages in groups of three to display correctly
+    msgs = msgs ++ ["Roll value: " <> Integer.to_string(special_roll)]
+    msgs = if special_roll == 6, do: msgs ++ ["It's super effective!"] ++ [player <> " used their special attack!"],
+      else: msgs ++ ["But it missed!"] ++ [player <> " used their special attack..."]
+
+    Map.put(game, :messages, msgs)
+  end
+
+  defp handle_specials(game, p1, p2) do
+    p1_special_roll = Map.get(p1, :specialRoll)
+    p2_special_roll = Map.get(p2, :specialRoll)
+    updated_game = game
+
+    if p1_special_roll != nil do
+      updated_game = update_messages(updated_game, p1.char, p1_special_roll, p1_special_roll != nil)
+    end
+
+    if p2_special_roll != nil do
+      updated_game = update_messages(updated_game, p2.char, p2_special_roll, p2_special_roll != nil)
+    end
+
+    updated_game
+  end
+
   # Determines who won the round and calculates the damage taken accordingly
   defp determine_winner(game) do
     players = Map.get(game, :players)
     p1 = Enum.at(players, 0)
     p2 = Enum.at(players, 1)
     p1Attack = Map.get(p1, :attack)
     p2Attack = Map.get(p2, :attack)
+    p2_special? = p2Attack == "Special"
+    special_activated? = p1Attack == "Special" or p2_special?
     p1Char = Map.get(p1, :char)
     p2Char = Map.get(p2, :char)
+    updated_game = game
     loser = nil
 
-    # Both chose the same attack, so no damage taken
-    if String.equivalent?(p1Attack, p2Attack) do
-      update_messages(game, p1Char, p1Attack, p2Char, p2Attack)
-    else
-      cond do
-        String.equivalent?(p1Attack, "Rock") ->
-          if String.equivalent?(p2Attack, "Paper"), do:
-            loser = 0,
-          else:
-            loser = 1
-        String.equivalent?(p1Attack, "Paper") ->
-          if String.equivalent?(p2Attack, "Rock"), do:
-            loser = 1,
-          else:
-            loser = 0
-        String.equivalent?(p1Attack, "Scissor") ->
-          if String.equivalent?(p2Attack, "Rock"), do:
-            loser = 0,
-          else:
+    if !String.equivalent?(p1Attack, p2Attack) do
```

Comment:
In the case where P1 and P2 both successfully use their specials in the same round, nothing would happen. Is this the behavior we want? It might be more fun if they both get to deal 40 damage to each other haha.

Reply:
I considered it as them both canceling out each other's special because we would have to then deal with the case of them both reaching a health of 0 lol

Reply:
Oh I didn't think about that. Nvm then.